src/evict/evict_file.c

```diff
@@ -108,8 +115,7 @@ __wt_evict_file(WT_SESSION_IMPL *session, int syncop)
 			 * connection close, and in other paths our caller
 			 * should be prepared to deal with this case.
 			 */
-			if (syncop == WT_SYNC_DISCARD &&
-			    page->modify != NULL &&
+			if (syncop == WT_SYNC_DISCARD && page->modify != NULL &&
 			    !__wt_txn_visible_all(session,
 			    page->modify->rec_max_txn))
 				WT_ERR(EBUSY);
```

Comment:
@michaelcahill, this might be worth breaking the two cases up, they're pretty different. Here's what it looks like:
```
case WT_SYNC_DISCARD:
        /*
         * Ordinary discard of the page, whether clean or dirty.
         * If we see a dirty page in an ordinary discard (e.g.,
         * from sweep), give up: an update must have happened
         * since the file was selected for sweeping.
         */
        if (__wt_page_is_modified(page))
                WT_ERR(EBUSY);
        /*
         * If the page contains an update that is too recent to
         * evict, stop.  This should never happen during
         * connection close, but in other paths our caller
         * should be prepared to deal with this case.
         */
        if (page->modify != NULL &&
            !__wt_txn_visible_all(session,
            page->modify->rec_max_txn))
                WT_ERR(EBUSY);

        __wt_evict_page_clean_update(session, ref);
        break;
case WT_SYNC_DISCARD_FORCE:
        /*
         * Forced discard of the page, whether clean or dirty.
         * If we see a dirty page in a forced discard, clean
         * the page, both to keep statistics correct, and to
         * let the page-discard function assert no dirty page
         * is ever discarded.
         */
        if (__wt_page_is_modified(page)) {
                page->modify->write_gen = 0;
                __wt_cache_dirty_decr(session, page);
        }
        /*
         * No page should ever contain an update that is too 
         * recent to evict.
         */
        WT_ASSERT(session, page->modify == NULL ||
            __wt_txn_visible_all(session,
            page->modify->rec_max_txn));

        F_SET(session, WT_SESSION_DISCARD_FORCE);
        __wt_evict_page_clean_update(session, ref);
        F_CLR(session, WT_SESSION_DISCARD_FORCE);
        break;
```